src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -653,5 +653,10 @@ public override IFileSystemObject GetFileSystemInfo(string fullPath, bool asDire
                 (IFileSystemObject)new DirectoryInfo(fullPath, null) :
                 (IFileSystemObject)new FileInfo(fullPath, null);
         }
+
+        public override string[] GetLogicalDrives()
+        {
+            return DriveInfoInternal.GetLogicalDrives();
```

Comment:
> I'm not seeing how the latter is better than the former.

And I'm not seeing the reverse. Even without needing to do it in System.Environment there is still the view from common- it's confusing to see just one platform specific file there. I still think it's far better from a repo comprehension standpoint to keep any implementation that has platform variants in one spot.

I know we have just one line for some of them now, but there is no guarantee that they will stay that way. There also isn't a guarantee on how many consumers there will be in the future. I'd rather see us be strict about the pattern in Common so we don't have auditing headaches as the repo grows. Without a strong argument to _not_ keep variants together in Common I can't see giving that benefit up. Right now the only argument I'm seeing is that it looks silly in FileSystem (and I totally agree it does)- but I don't think is enough to trump a clearer Common.